GitCommands/Git/GitTagController.cs

```diff
@@ -0,0 +1,111 @@
+﻿using GitUIPluginInterfaces;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace GitCommands.Git
+{
+    public interface IGitTagController
+    {
+        /// <summary>
+        /// Create the Tag depending on input parameter.
+        /// </summary>
+        /// <param name="revision">Commit revision to be tagged</param>
+        /// <param name="tagName">Name of tag</param>
+        /// <param name="force">Force parameter</param>
+        /// <param name="operationType">Available tag operationType (Lightweight, Annotate, Sign with defaul key, Sign with specific key)</param>
+        /// <param name="tagMessage">Tag Message</param>
+        /// <param name="keyId">Specific Key ID to be used instead of default one</param>
+        /// <returns>Output string from RunGitCmd.</returns>
+        string CreateTag(string revision, string tagName, bool force, GitTagController.TagOperation operationType = GitTagController.TagOperation.Lightweight, string tagMessage = "", string keyId = "");
+    }
+
+
+    public class GitTagController : IGitTagController
+    {
+        private IGitModule _module;
+
+
+        public enum TagOperation
+        {
+            Lightweight = 0,
+            Annotate = 1,
+            SignWithDefaultKey = 2,
+            SignWithSpecifiKey = 3
+        };
```

Comment:
All those value are needed by FormCreateTag to know if "specific key id" shall be enabled or not.
You'll see in my next commit.